proxy/http/HttpSM.cc

```diff
@@ -273,7 +370,7 @@ HttpVCTable::cleanup_all()
 #define STATE_ENTER(state_name, event)                                                 \
   {                                                                                    \
     /*ink_assert (magic == HTTP_SM_MAGIC_ALIVE); */ REMEMBER(event, reentrancy_count); \
-    SMDebug("http", "[%s, %s]", #state_name, HttpDebugNames::get_event_name(event));   \
+    SMDebugHttp("[%s, %s]", #state_name, HttpDebugNames::get_event_name(event));       \
```

Comment:
It's sad that I can't see the actual tag name, although that might have already been discussed when the new interface was proposed. This one is ok, but "ip-allow" is tricky because it uses "-" where most other tags use "_".

Reply:
@SolidWallOfCode did you pick the tag name "ip-allow"?  Do you care if I change it to "ip_allow"?

I could also change to a naming pattern where, for example, SMDebug_http_trans (rather than SMDebugHttpTrans) corresponds to the tag "http_trans".

Reply:
If all debug tags were valid C++ identifiers, then I wouldn't need the SMDebugXxx() macros.  This:
```
SMDebug("http_trans", "Unable to accelerate range request, fallback to transform");
```
would instead become this:
```
SMDebug(http_trans, "Unable to accelerate range request, fallback to transform");
```
The implementation would use either the concatenation or stringization macro operators:
```
#if defined(Dbg)

namespace
{
DbgCtl dbgCtl_http_trans;
// ...
}

#define SMDebug(tag, fmt, ...) SpecificDbg(debug_on, dbgCtl_ # tag, "[%" PRId64 "] " fmt, sm_id, ##__VA_ARGS__)

#else

#define SMDebug(tag, fmt, ...) SpecificDbg(debug_on, #tag, "[%" PRId64 "] " fmt, sm_id, ##__VA_ARGS__)

#endif
```

Reply:
> If all debug tags were valid C++ identifiers...

I like the concatenation/stringization macro idea. It would be confusing a little since old Debug() requires quotations though, I prefer it rather than other naming patterns.

Also, somebody told me that he prefer to have "Debug" at the end of macro names because that enables you to find (grep) Debug lines easily, when I added macros for HTTP/2 or 3.